File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/visiting/ListEvaluator.java
package org.opencds.cqf.cql.engine.elm.visiting;

import org.hl7.elm.r1.Expression;
import org.hl7.elm.r1.List;
import org.opencds.cqf.cql.engine.execution.CqlEngineVisitor;
import org.opencds.cqf.cql.engine.execution.State;

import java.util.ArrayList;

public class ListEvaluator {

    public static Object internalEvaluate(List list, State state, CqlEngineVisitor visitor) {
        ArrayList<Object> result = new ArrayList<Object>();
        for (Expression element : list.getElement()) {
            Object obj = visitor.validateOperand(visitor.visitExpression(element, state));
            result.add(obj);
        }
        return result;
    }
}

File: engine/src/commonMain/kotlin/org/opencds/cqf/cql/engine/elm/executing/ListEvaluator.kt
package org.opencds.cqf.cql.engine.elm.executing

import org.cqframework.cql.elm.visiting.ElmLibraryVisitor
import org.opencds.cqf.cql.engine.execution.State
import org.opencds.cqf.cql.engine.runtime.List
import org.opencds.cqf.cql.engine.runtime.Value
import org.opencds.cqf.cql.engine.runtime.toCqlList

object ListEvaluator {
    fun internalEvaluate(
        list: org.hl7.elm.r1.List?,
        state: State?,
        visitor: ElmLibraryVisitor<Value?, State?>,
    ): List {
        val result = mutableListOf<Value?>()
        for (element in list!!.element) {
            val obj = visitor.visitExpression(element, state)
            result.add(obj)
        }
        return result.toCqlList()
    }
}
